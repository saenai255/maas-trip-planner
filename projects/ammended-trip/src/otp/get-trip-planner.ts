import Axios from 'axios';
import { Coordinates } from '../dto/coordinates';
import config from '../config';
import TripPlannerResponse from './dto/trip-planner-response';

async function getTripPlanner(from: Coordinates, to: Coordinates): Promise<TripPlannerResponse> {
    const now = new Date();
    const additionalParams: Record<string, string | number | boolean> = {
        time: now.getHours() + ':' + now.getMinutes(),
        date: now.getDate() + '-' + (now.getMonth() + 1) + '-' + now.getFullYear(),
        mode: 'TRANSIT',
        arriveBy: false,
        wheelchair: false,
        showIntermediateStops: true,
        locale: 'en',
    };

    const params = new URLSearchParams({
        fromPlace: from.join(','),
        toPlace: to.join(','),
        ...additionalParams,
    });

    const result = await Axios.get<TripPlannerResponse>(`${config.otp.url}/otp/routers/default/plan?${params.toString()}`, {
        headers: {
            'x-api-key': config.otp.apiKey,
        },
    });

    return result.data;
}

export { getTripPlanner };
