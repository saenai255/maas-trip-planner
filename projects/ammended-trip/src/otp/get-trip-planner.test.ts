import Axios from 'axios';
import { getTripPlanner } from './get-trip-planner';
import TripPlannerResponse from './dto/trip-planner-response';

jest.mock('axios', () => ({
    default: {
        get: jest.fn(),
    },
}));

describe('getTripPlanner', () => {
    const axiosGetMock = Axios.get as jest.MockedFunction<typeof Axios.get>;

    it('should return routes', async () => {
        const expectedResponse: TripPlannerResponse = {} as any;
        axiosGetMock.mockResolvedValueOnce({
            data: expectedResponse,
        } as any);

        const result = await getTripPlanner([0, 0], [0, 0]);
        expect(axiosGetMock).toHaveBeenCalledTimes(1);
        expect(result).toBe(expectedResponse);
    });

    it('should throw an error if the request fails', async () => {
        axiosGetMock.mockRejectedValueOnce(new Error('Request failed'));

        await expect(getTripPlanner([0, 0], [0, 0])).rejects.toThrowErrorMatchingSnapshot();
    });
});
