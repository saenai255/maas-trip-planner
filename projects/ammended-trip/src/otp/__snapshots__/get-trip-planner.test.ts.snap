// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTripPlanner should throw an error if the request fails 1`] = `"Request failed"`;
